dai_cadcad/flip.py

```diff
@@ -0,0 +1,116 @@
+""" Flip Module
+    Class-based representation of the Flip smart contract
+    (contains only what is necessary for the simulation)
+"""
+
+from dai_cadcad.util import require
+
+
+class Flipper:
+    ADDRESS = ""
+
+    vat = None
+    beg = None
+    ttl = None
+    tau = None
+    kicks = None
+    bids = {}
+    ilk = None
+
+    def __init__(self, vat, beg, ttl, tau, kicks, bids, ilk):
+        """"""
+
+        self.ADDRESS = f"flipper_{ilk}"
+
+        self.vat = vat
+
+        self.beg = beg
+        self.ttl = ttl
+        self.tau = tau
+        self.kicks = kicks
+        self.bids = bids
+        self.ilk = ilk
+
+    def kick(
+        self, usr, gal, tab, lot, bid, end
+    ):  # TODO: should these be instance attribute?
+        """Kicks off a new Flip auction."""
+
+        self.kicks += 1
+        bid_id = self.kicks
+
+        self.bids[bid_id] = {
+            "bid": bid,
+            "lot": lot,
+            "guy": "cat",
+            "tic": 0,
+            "end": end,
+            "usr": usr,
+            "gal": gal,
+            "tab": tab,
+        }
+
+        self.vat.flux(self.ilk, "cat", "flipper_eth", lot)
```

Comment:
hmm, if `self.ilk` is a rich object/dict, then we'd have to use it's ID in the call to `flux` here. if it's just an ilk ID, can we rename the variable to `ilk_id` everywhere appropriate